examples/drupal_loadtest.rs

```diff
@@ -81,19 +81,27 @@ fn main() {
 }
 
 /// View the front page.
-async fn drupal_loadtest_front_page(client: &mut GooseClient) {
-    let mut response = client.get("/").await;
+async fn drupal_loadtest_front_page(client: &GooseClient) {
+    let mut response = client.get("/", None).await;
```

Comment:
@LionsAd Now that `GooseClient` is immutable, I had to add a new way to set per-request names. Ultimately this meant adding a new parameter to all requests, which is an optional name. Is this too unwieldy? (ie, you either pass in `None` if you're not setting a custom request name, or you pass in `Some("request name")` if you are).

Another idea I had was to make two versions of each request type, for example:
`fn get(path)` and `fn get_named(path, name)`. Perhaps this is cleaner, as you don't have to pass in `None` when you're not naming requests (and I expect normally people will _not_ be naming requests).

Reply:
I split `.get()` and `.get_named()` as I find it has better ergonomics, and `.get()` is already just a convenience wrapper. (And the same for `.post()`, `.head()`, and `.delete()`.) This reduces the size of this PR, as it requires fewer changes when writing most load tests.